redis/connection.py

```diff
@@ -968,7 +976,33 @@ def _connect(self):
         sslsock = context.wrap_socket(sock, server_hostname=self.host)
         if self.ssl_validate_ocsp is True and CRYPTOGRAPHY_AVAILABLE is False:
             raise RedisError("cryptography is not installed.")
-        elif self.ssl_validate_ocsp is True and CRYPTOGRAPHY_AVAILABLE:
+
+        if self.ssl_ocsp_context is not None and self.ssl_validate_ocsp is not False:
+            raise RedisError(
+                "Either an OCSP context or validation may be specified " "but not both."
+            )
+
+        # validation for the stapled case
+        if self.ssl_ocsp_context:
+            import OpenSSL
+
+            from .ocsp import ocsp_staple_verifier
+
+            self.ssl_ocsp_context.set_ocsp_client_callback(
+                ocsp_staple_verifier,
+                self.ssl_ocsp_expected_cert,
+            )
+
+            #  need another socket
+            con = OpenSSL.SSL.Connection(self.ssl_ocsp_context, socket.socket())
+            con.request_ocsp()
+            con.connect((self.host, self.port))
+            con.do_handshake()
+            con.shutdown()
```

Comment:
isn't this actually closing the connection?

Reply:
Correct. the OCSP validation occurs on a different socket. I was unable to reusse the same socket for the internal ssl implementation and PyOpenSSL. 

In my current thinking - future versions of redis-py (5?) I'd like to see us operate more similarly to node, and have the user supply the context.  But I'd love feedback here.